### PR TITLE
doc(connect-explorer): step by step tutorial to build webextension using @trezor/connect-webextension

### LINKED ISSUE #13604: Create tutorial on how to implement Connect into 3rd party webextension
Exact place in explorer tbd.

### PR DESCRIPTION
**Pull request by karliatto**
## Description

Adding a step by step tutorial to create a webextension that uses `@trezor/connect-webextension` with the help of TypeScript and Webpack.

The path in connect-explorer has to be decided still.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/13604

## Build

https://dev.suite.sldev.cz/connect/doc/tutorial-for-connect-webextension/

### CODE REVIEW

File: packages/connect-explorer/src/pages/blog.mdx [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+                // Return true to indicate you want to send a response asynchronously
+                return true;
+            }
+        });
+
+    });
+
+    ````
+
+    So far your directoy looks like:
```
Comment (martykan) [inline packages/connect-explorer/src/pages/blog.mdx]: little typo

File: packages/connect-explorer/src/pages/blog.mdx [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+import styled from 'styled-components';
+
+import { spacingsPx } from '@trezor/theme';
+import { Card as TrezorCard } from '@trezor/components';
+export const SectionCard = styled(TrezorCard)`
+    margin-bottom: ${spacingsPx.xl};
+`;
+
+# Create a webextension that interacts with Trezor devices using Typescriptand Webpack
```
Comment (martykan) [inline packages/connect-explorer/src/pages/blog.mdx]: typo "Typescriptand"

File: packages/connect-explorer/src/pages/blog.mdx [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+    1. Create a New Directory for Your Project: Open your terminal
+    and create a new directory for your web extension project.
+
+    ```bash
+    mkdir my-trezor-extension
+    cd my-trezor-extension
+    ```
+
+    2. Initialize a New Yarn Project: Run the following command to create a
```
Comment (martykan) [inline packages/connect-explorer/src/pages/blog.mdx]: Yarn is not actually used

File: packages/connect-explorer/src/pages/blog.mdx [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+
+    ````
+
+    So far your directoy looks like:
+
+    ```
+    node_modules
+    src
+        service-worker.ts
+    pacakge-lock.json
```
Comment (martykan) [inline packages/connect-explorer/src/pages/blog.mdx]: typo

**Review by martykan — COMMENTED** [review]
Found some small typos, but overall the article looks good. 

Let's also discuss how to handle the navigation, there should probably be some index page that would list the articles.

I'm looking here for inspiration 
https://swr.vercel.app/blog
https://github.com/vercel/swr-site/blob/main/components/blog-index.js

 But maybe we can get to that later when we add other articles.

File: packages/connect-explorer/src/pages/blog.mdx (hunk 1 above)
Comment (karliatto) [reply packages/connect-explorer/src/pages/blog.mdx]: https://github.com/trezor/trezor-suite/pull/14043/commits/695cdede30d0d770489740225a9c0c55e4f6e96e

File: packages/connect-explorer/src/pages/blog.mdx (hunk 2 above)
Comment (karliatto) [reply packages/connect-explorer/src/pages/blog.mdx]: https://github.com/trezor/trezor-suite/pull/14043/commits/695cdede30d0d770489740225a9c0c55e4f6e96e

File: packages/connect-explorer/src/pages/blog.mdx (hunk 3 above)
Comment (karliatto) [reply packages/connect-explorer/src/pages/blog.mdx]: https://github.com/trezor/trezor-suite/pull/14043/commits/695cdede30d0d770489740225a9c0c55e4f6e96e

File: packages/connect-explorer/src/pages/blog.mdx (hunk 4 above)
Comment (karliatto) [reply packages/connect-explorer/src/pages/blog.mdx]: https://github.com/trezor/trezor-suite/pull/14043/commits/695cdede30d0d770489740225a9c0c55e4f6e96e

**Comment by karliatto** [comment]
> 
> Let's also discuss how to handle the navigation, there should probably be some index page that would list the articles.
> 

I think that is a good idea, I am adding it. Thanks for the inspiration!

**Comment by karliatto** [comment]
@Hannsek it would be nice to get some feedback about where to put it and the whole concept. I will leave as it is for now till we agree on how to proceed.

File: packages/connect-explorer/src/pages/blog/webextension-ts-webpack.mdx [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+    tsconfig.json
+    ```
+
+</SectionCard>
+
+<SectionCard>
+    ## Create tab that will communicate with service-worker
+    The library `@trezor/connect-webextension` is running in the service-worker because it allows the webextension to have
+    a persistent reference.
+    So we if want our front-end part of the webextension to be able to communicate with service-worker one way we can do it
```
Comment (Hannsek) [inline packages/connect-explorer/src/pages/blog/webextension-ts-webpack.mdx]:
```suggestion
    So if we want our front-end part of the webextension to be able to communicate with service-worker one way we can do it
```

File: packages/connect-explorer/src/pages/blog/webextension-ts-webpack.mdx [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+        "scripts": {
+            "build": "webpack --mode production"
+        },
+        ...
+    ```
+
+</SectionCard>
+
+<SectionCard>
+    ## Your directoy should look like:
```
Comment (Hannsek) [inline packages/connect-explorer/src/pages/blog/webextension-ts-webpack.mdx]:
```suggestion
    ## Your directory should look like:
```

**Comment by Hannsek** [comment]
Thank you very much for this. Overall great job. I have one small nit:
- some names of packages or files are not differentiated, but some are
image
image
.

I'll try to build the example according to the tutorial sometimes next week.

I am not a fan of having it under the "blog" item. We may rename it to "Guides" or something like that.

Ultimately I'd like to move _environment cards_ from the welcome screen to dedicated pages. There should be the whole environment card with the "quick start" section but also there can be a section with "step-by-step tutorial on how to build webextension example with connect-webextension package". Let's discuss next week on standup.

**Comment by Hannsek** [comment]
Ok, let's put the link under the "open tutorial" button under the "Learn step-by-step" paragraph in web-extension tile:
image

**Comment by karliatto** [comment]
> Ok, let's put the link under the "open tutorial" button under the "Learn step-by-step" paragraph in web-extension tile.

@Hannsek I updated it as you suggested, let me know if you still think some change can improve it.

https://dev.suite.sldev.cz/connect/doc/tutorial-for-connect-webextension/

**Comment by Hannsek** [comment]
Great, let's just fix the spaces above and below the button.
image


What do you think about adding the `Learn step by step` paragraph there? I think this can be useful to point users to the tutorial.

Also, can we rename the file? To e.g. `@trezor/connect-webextension implementation tutorial`?
image

**Comment by karliatto** [comment]
> `Learn step by step`

@Hannsek Something like:

image


:question:

**Comment by Hannsek** [comment]
Yes, something like that. Take a look at the picture [above](https://github.com/trezor/trezor-suite/pull/14043#issuecomment-2348336839). There are two paragraph names (ABOUT and LEARN STEP-BY-STEP). In the LEARN STEP-BY-STEP paragraph, there can be this text:

"Learn how to integrate the @trezor/connect-webextension package into a Chrome extension, allowing secure connection to Trezor hardware wallets. By the end, you'll have a fully functional web extension."

**Comment by karliatto** [comment]
> Learn how to integrate the @trezor/connect-webextension package into a Chrome extension, allowing secure connection to Trezor hardware wallets. By the end, you'll have a fully functional web extension.

@Hannsek Sure, I did not realize of that detail, thanks for pointing it out. What about?

image
:question:

**Comment by Hannsek** [comment]
Great! One more thing. Please add "About" to all tiles (connect-web and connect). 

And that would be it.

**Comment by karliatto** [comment]
> Great! One more thing. Please add "About" to all tiles (connect-web and connect). Can we also change the font so that it is aligned with "examples" title font? 

@Hannsek I added the `About` header to all the tiles.
Rergarding the icon in the left section for the Guides it looks like the nextra framework does not add icons to those elements when they have sub elements as the "Guides" section does. So it would require changes in the design system to do that, and we can do it as a follow up if really required.

**Comment by Hannsek** [comment]
Let's not do it then.

@martykan pointed out yesterday that we should change the name once more to just "Webextension Implementation Tutorial"
image

Also, I've tried it now and the button leading to the actual tutorial in the webextension tile doesn't work.

**Comment by karliatto** [comment]
> Webextension Implementation Tutorial

@Hannsek sure, I updated for that name and fixed the button leading to the tutorial, thanks!

**Comment by Hannsek** [comment]
Thanks. When I click the "show tutorial" button, It opens up in the new browser tab, which is not ideal. Can we do something with it so it is opened in the same tab? 🤔

**Comment by karliatto** [comment]
> Thanks. When I click the "show tutorial" button, It opens up in the new browser tab, which is not ideal. Can we do something with it so it is opened in the same tab? 🤔

@Hannsek Sure, that makes sense. Changed https://dev.suite.sldev.cz/connect/doc/tutorial-for-connect-webextension/

**Comment by Hannsek** [comment]
I think we are ready to merge. Thank you Carlos!

**Comment by martykan** [comment]
I just added the icon to the menu and made some design tweaks to the index page

**Comment by karliatto** [comment]
> I just added the icon to the menu and made some design tweaks to the index page

Cool, looks nice. I rebased it, so I guess it is ready.

**Review by martykan — APPROVED** [review]

**Pull request merged**